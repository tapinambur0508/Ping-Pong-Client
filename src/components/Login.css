@import url('https://fonts.googleapis.com/css?family=Poppins');
#login .wrapper {
  max-width: 400px;
  margin: 50px auto;
  padding: 30px;
  border-radius: 10px;
  background: #ffffff;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
}

#login .wrapper .heading {
  text-align: center;
  font-family: 'Poppins', sans-serif;
}

#login .wrapper button.kep-login-facebook {
  margin: 10px 0;
  border-radius: 50px;
  text-indent: 10px;
  border: none;
  width: 100%;
}