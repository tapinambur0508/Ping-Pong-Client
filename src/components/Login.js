import React from 'react';
import FacebookLogin from 'react-facebook-login';
import withRouter from 'react-router-dom/withRouter';
import axios from 'axios';

import AuthService from '../services/auth';

import config from '../config.json';

import './Login.css';

const Login = props => {
  const facebookResponse = response => {
    axios({
      method: 'POST',
      url: 'https://ping-pong-main-server.herokuapp.com/api/oauth/facebook',
      data: {
        access_token: response.accessToken
      },
      headers: {
        'Content-Type': 'application/json'
      }
    })
      .then(({ data }) => {
        const authService = new AuthService();

        authService.token = data['access_token'];
        props.history.push('/');
      })
      .catch(err => console.log(err));
  };

  return (
    <section id="login">
      <div className="container">
        <div className="wrapper">
          <h1 className="heading">Ping-Pong</h1>
          <FacebookLogin
            appId={config.FACEBOOK_APP_ID}
            autoLoad={false}
            fields="name,email,picture"
            textButton="Continue with Facebook"
            callback={facebookResponse} />
        </div>
      </div>
    </section>
  );
}

export default withRouter(Login);
