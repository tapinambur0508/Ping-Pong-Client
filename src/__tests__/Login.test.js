import React from 'react';
import Login from '../components/Login';

describe('Login', () => {
  it('should be defined', () => {
    expect(Login).toBeDefined();
  });
});
