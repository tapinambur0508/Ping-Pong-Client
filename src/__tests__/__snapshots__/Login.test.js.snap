// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Home Component matches the snapshot 1`] = `
<section
  id="login"
>
  <div
    className="container"
  >
    <div
      className="wrapper"
    >
      <h1
        className="heading"
      >
        Ping-Pong
      </h1>
      <span
        style={
          Object {
            "opacity": 0.6,
            "transition": "opacity 0.5s",
          }
        }
      >
        <button
          className="kep-login-facebook metro"
          onClick={[Function]}
          type="button"
        >
          Continue with Facebook
        </button>
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": Array [
                Array [
                  10,
                  ".kep-login-facebook{font-family:Helvetica,sans-serif;font-weight:700;-webkit-font-smoothing:antialiased;color:#fff;cursor:pointer;display:inline-block;font-size:calc(.27548vw + 12.71074px);text-decoration:none;text-transform:uppercase;transition:background-color .3s,border-color .3s;background-color:#4c69ba;border:calc(.06887vw + .67769px) solid #4c69ba;padding:calc(.34435vw + 13.38843px) calc(.34435vw + 18.38843px)}.kep-login-facebook.small{padding:calc(.34435vw + 3.38843px) calc(.34435vw + 8.38843px)}.kep-login-facebook.medium{padding:calc(.34435vw + 8.38843px) calc(.34435vw + 13.38843px)}.kep-login-facebook.metro{border-radius:0}.kep-login-facebook .fa{margin-right:calc(.34435vw + 3.38843px)}",
                  "",
                ],
              ],
            }
          }
        />
      </span>
    </div>
  </div>
</section>
`;
